### ISSUE TITLE
Didn't Install Pixel Ambient Services and causes Play Services to crash

### ISSUE THREAD

**Issue by MKCOOL142** [KieronQuinn/AmbientMusicMod#33]
So I tried to follow the instructions just like you mentioned, Installed the Magisk module, enabled the XPosed module and whitelisted the apps, but at that point I realized that I don't have the Pixel Ambient Services (I assume the module is supposed to install it themselves because you no where mention which version is even supported). It also made my play services continuously crash. I don't know if this is because my Phone (OnePlus 7 Pro) was listed as incompatible but I don't think that's the cause. Please let me know if this is my fault or if there's something I can do to fix it, thank you

![Screenshot_20211205-203406](https://user-images.githubusercontent.com/40364718/144761536-69702f3c-b1e6-45e8-82ce-34cfc61fb18a.jpg)
![Screenshot_20211205-204053](https://user-images.githubusercontent.com/40364718/144761539-1b0ffc25-ac6f-4126-8315-f966efc918aa.jpg)

**Comment by KieronQuinn** [comment]
Is it not installed at all, not even appearing in the system settings under apps? You might want to reinstall the module.

**Comment by MKCOOL142** [comment]
Nope, it's not there either and yes, I just tried for a second time, can I manually install the apks or do they have to be system apps?

**Comment by KieronQuinn** [comment]
It has to be a system app, but installing it manually will tell us whether something weird is going on. 

Try to install the APK (it's in the zip), if it shows "Update" then you have a very broken system, if it fails to install then proceed to step 2. If it installs fine then god knows what's going on.

If it fails with "App not installed", you'll have to install it with ADB, or use an app from the store that shows the reason (I usually use the APKmirror app for this), which will hopefully tell us what's wrong.

**Comment by MKCOOL142** [comment]
I have to correct myself on my last comment, the magisk module was disabled while I made those screenshots because the majority of apps not working and play services giving me errors wasn't a pleasant experience, so I'll check the folder again but excuse me, I don't really know that much about how magisk works so my those folders are supposed to exist even with the module disabled, about manually installing the app... It did infact install fine
![Screenshot_20211205-232031](https://user-images.githubusercontent.com/40364718/144766343-41cde869-7187-41ad-be1f-8da159a15835.jpg)

**Comment by KieronQuinn** [comment]
If the module was disabled, then the app wouldn't be installed, that is intended behaviour. The folders wouldn't exist either as the bind mount won't have happened.

**Comment by MKCOOL142** [comment]
Okay so since I manually installed the app my play services don't crash anymore and the app sorta works but I don't think that the ambient services themselves are functional, my /system/product/priv-app/ folder now contains the com.google.intelligence.sense folder too, except all the folders are now empty, I don't know if this intended
![Screenshot_20211205-232910](https://user-images.githubusercontent.com/40364718/144766582-27f309fa-6f53-45ca-b7e2-54eb3c29747a.jpg)

**Comment by KieronQuinn** [comment]
That is not intended, no. Can you send the module zip that Ambient Music Mod generated?

**Comment by MKCOOL142** [comment]
https://drive.google.com/file/d/1ud69WYB8kckGHRwMRmvZ6zR6sfnmyw5n/view?usp=drivesdk

Sorry that it took longer, I got terribly slow wifi

**Comment by KieronQuinn** [comment]
Thanks. The APK is in that zip, so I don't know why Magisk isn't mounting it. Does it appear in /data/adb/modules/AmbientMusicMod/system/product/priv-app/com.google.intelligence.sense?

**Comment by MKCOOL142** [comment]
Yes, it does appear there

**Comment by KieronQuinn** [comment]
You may need this module, possibly: https://github.com/Magisk-Modules-Repo/overlayfix

It's available in the Magisk app if you're **not** using Magisk Canary.

**Comment by MKCOOL142** [comment]
Thank you so much, that did actually work, since it's now probably working fine (will have to try playing actual music tomorrow), I wanted to give you a small feedback but didn't wanna make extra threads for that:
- I think you should link the overlay fix just in case others experience similar issues
- you should fix the wording in the troubleshooting segment of the app if the test recognition fails because it only says "android", where I assume it's supposed to be "system Framework", it also doesn't mention that newer android version call "Device personalization Services" "Android system intelligence" instead, I only knew that the system Framework was required by finding a different closed issue where you mentioned the list of apps that have to be whitelisted
![Screenshot_20211206-002118__01](https://user-images.githubusercontent.com/40364718/144768496-c4a9252b-f7f8-4e35-96c4-ec979b5cb53b.jpg)

**Comment by KieronQuinn** [comment]
I wasn't aware the overlay fix was still needed, I stopped needing it on my old 7TP over a year ago (I built the module on that device), maybe it's still needed on some OP devices. 

As for the wording, it does need updating but that needs an app update. EdXposed calls the system "android" (which is its package name), lsposed has special handling to call it Android System Framework. It'll happen eventually.
